Mathlib/Tactic/Linter/MinImports.lean

```diff
@@ -85,15 +110,21 @@ def minImportsLinter : Linter where run := withSetOptionIn fun stx => do
         logWarningAt ((impMods.find? (·.isOfKind `import)).getD default)
           m!"-- missing imports\n{"\n".intercalate withImport.toList}"
     let id ← getId stx
-    let newImports := getIrredundantImports (← getEnv) (← getAllImports stx id)
+    let newImports := getIrredundantImports env (← getAllImports stx id)
     let tot := (newImports.append importsSoFar)
-    let redundant := (← getEnv).findRedundantImports tot.toArray
+    let redundant := env.findRedundantImports tot.toArray
     let currImports := tot.diff redundant
     let currImpArray := currImports.toArray.qsort Name.lt
     if currImpArray != #[] &&
        currImpArray ≠ importsSoFar.toArray.qsort Name.lt then
-      minImportsRef.modify fun _ => currImports
-      Linter.logLint linter.minImports stx m!"Imports increased to\n{currImpArray}"
+      let newCumulImps := (importsBelow env.importGraph tot).size
+      minImportsRef.set {minImps := currImports, impsSize := newCumulImps}
+      let new := currImpArray.filter (!importsSoFar.contains ·)
+      let redundant := importsSoFar.toArray.filter (!currImports.contains ·)
```

Comment:
What function is missing that causes you to need `toArray`? Let's just add this function.

Reply:
I think that it is the `filter` and then the fact that I want to print it anyway.